src/components/CreateAnalysis/CreateAnalysisDetail.tsx

```diff
@@ -10,31 +9,31 @@ import SelectionStudy from "./SelectionStudy";
 import FileLookup from './FileLookup';
 import Error from "../../shared/error";
 import { PageSection, PageSectionVariants } from "@patternfly/react-core";
+import { DcmImage } from "../../context/reducers/dicomImagesReducer";
 
 interface CreateAnalysisDetailProps {
   setIsExpanded: Dispatch<SetStateAction<boolean>>,
   submitAnalysis: () => void
 }
 
 const CreateAnalysisDetail: React.FC<CreateAnalysisDetailProps> = ({setIsExpanded, submitAnalysis}) => {
-  const { state: { dcmImages, createAnalysis }, dispatch } = useContext(AppContext);
-  const { selectedStudyUIDs } = createAnalysis;
+  const { state: { createAnalysis, dcmImages }, dispatch } = useContext(AppContext);
   const [isXray, setIsXray] = useState(false);
+  const [patientName, setPatientName] = useState('');
+  const [patientBirthdate, setPatientBirthdate] = useState('');
+  const [patientSex, setPatientSex] = useState('');
 
   useEffect(() => {
-    const patientInfo = CreateAnalysisService.extractPatientPersonalInfo(dcmImages?.allDcmImages[0])
-    dispatch({
-      type: CreateAnalysisTypes.Update_patient_personal_info,
-      payload: {
-        ...patientInfo
-      }
-    })
-  }, [dcmImages, dispatch]);
+    const images: DcmImage[] = CreateAnalysisService.returnAllImagesInOneStudy(dcmImages?.filteredDcmImages, createAnalysis.currSelectedStudyUID);
```

Comment:
By using `returnAllImagesInOneStudy`, we are making the current user info dependent on the `currSelectedStudyUID`. Although in practice, the user info in the DICOM across different studies shouldn't be different, there isn't a need here to update the user info every time the selected study changes.

Also, is there a reason why we're using `returnAllImagesInOneStudy` instead of `extractPatientPersonalInfo`?

Reply:
I decided to use returnAllImagesInOneStudy to make it more consistent with how the age is read at the bottom (the second patient age section, beside performed station). This section also uses the same way of reading in data, where it's dependent on the study and changes based on the study. so just to make the code more consistent and bug-proof, i thought of doing it like this to maintain that the two values would always be in sync

But you're right in that if the DICOM data is good, this shouldn't be a worry

As for extractpatientpersonalInfo, I removed that function because it was only being used in one location, and it was needlessly complicated, where it took in an object, read it in, returned another object and that object was then read again. Instead, it's easier to just directly read the object and access the desired fields we want